### ISSUE TITLE
Feature request: save as mime-type

### ISSUE THREAD

**Issue by wshager** [eXist-db/eXide#101]
When storing (text) documents of unkown mime-type in eXide, eXist complains about "content not allowed in prolog", i.e. not valid XML. Perhaps this should be addressed as eXist bug, but it could be solved in eXide by either retrieving the "correct" mime-type (not XML) or adding the option to save to a custom mime-type.

**Comment by joewiz** [comment]
I'm not seeing this behavior when I create a new text document using File > New > Type: Other.  What steps do you follow to see this error?

**Comment by wshager** [comment]
@joewiz  Open the file again and try to save it.

**Comment by joewiz** [comment]
@wshager I saved my file to /db/test.txt, reopened it, modified the file, and saved it.  No errors or indications of a problem.  I did the same with a new file without the .txt file extension - /db/blah - and got the same results.  I monitored the PUT requests, and eXide correctly set the mime-type via the Content-Type header: `Content-Type: text/text`.  So I'm afraid I'm not seeing what you are.  I'm using a current build of the develop branch, and the current eXide.

**Comment by wshager** [comment]
@joewiz Please read the original post: I meant _unknown_ mime-type. Adding text in brackets may have been misleading.

**Comment by joewiz** [comment]
I can't reproduce the error you're seeing.  Please provide steps to reproduce.

**Comment by wshager** [comment]
@joewiz perhaps this has to do with the version of eXist?

**Comment by joewiz** [comment]
Could be, but frankly that's speculation until you provide steps to reproduce the issue on your system.

**Comment by wshager** [comment]
@joewiz fair enough. I couldn't reproduce it either until I uploaded the file again using the manager. So the problem is in the storing by the db manager.

**Comment by wshager** [comment]
The default mime-type in xmldb:store seems to be application/xml. This probably should be addressed in eXist.

**Comment by joewiz** [comment]
@wshager Please, just a few steps:
1. ___
2. ___

**Comment by wshager** [comment]
- upload file.bla using eXide db manager
- open file.bla in eXide
- save file.bla

**Comment by dizzzz** [comment]
@wshager that helps; since eXist-db is an Native XML database, by design when a document cannot be recognized, there is an attempt to write it as an XML file.

**Comment by wshager** [comment]
@dizzzz ok agreed. So perhaps it would nice to have an option to add custom mime-types.

**Comment by dizzzz** [comment]
There is something alike on database level; I am not aware how the relation is with eXide though.  I did not verify it manually but it might help you.

``` xml
<mime-types default-mime-type="foo/bar" default-resource-type="binary" 
xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
xsi:noNamespaceSchemaLocation="schema/mime-types.xsd">    
</mime-types>
```

**Comment by wshager** [comment]
@dizzzz ah yes of course, that ought to work. I'll give it a try.

**Comment by joewiz** [comment]
I'll label this as a request for enhancement. Are you still interested, @wshager?

In the meantime, the mime type of any stored resource can be modified via eXide's File Manager > Properties pane:

> <img width="603" alt="screen shot 2017-03-12 at 5 07 36 pm" src="https://cloud.githubusercontent.com/assets/59118/23835944/706eb184-0746-11e7-801a-cb264435dfef.png">
